packages/number-field/test/number-field.test.js

```diff
@@ -0,0 +1,884 @@
+
+        decreaseButton.click();
+
+        expect(numberField.value).to.be.equal('0');
+      });
+    });
+
+    describe('max is defined and min is undefined', () => {
+      it('should set value to the closest to the max value when max is below zero and increaseButton is clicked', () => {
+        // -19 cannot be equally divided by 6
+        // The closest is -24, cause with the next stepUp it will become -18
+        numberField.max = -19;
+        numberField.step = 6;
+
+        increaseButton.click();
+
+        expect(numberField.value).to.be.equal('-24');
+
+        // Check with max that can be equally divided
+        numberField.value = '';
+        numberField.max = -18;
+        numberField.step = 6;
+
+        increaseButton.click();
+
+        expect(numberField.value).to.be.equal('-18');
+      });
+
+      it('should set value to max when max is below zero and decreaseButton is clicked', () => {
+        numberField.max = -19;
+        numberField.step = 6;
+
+        decreaseButton.click();
+
+        expect(numberField.value).to.be.equal('-19');
+      });
+
+      it('should set value to 0 when max is zero and increaseButton is clicked', () => {
+        numberField.max = 0;
+        numberField.step = 6;
+
+        increaseButton.click();
+
+        expect(numberField.value).to.be.equal('0');
+      });
+
+      it('should set value to the first step value above zero when max is above zero and increaseButton is clicked', () => {
+        numberField.max = 19;
+        numberField.step = 6;
+
+        increaseButton.click();
+
+        expect(numberField.value).to.be.equal('6');
+      });
+
+      it('should set value to the first step value below zero when max is above zero and decreaseButton is clicked', () => {
+        numberField.max = 19;
+        numberField.step = 6;
+
+        decreaseButton.click();
+
+        expect(numberField.value).to.be.equal('-6');
+      });
+
+      it('should set value to the first step value below zero when max is above zero and decreaseButton is clicked', () => {
+        numberField.max = 0;
+        numberField.step = 6;
+
+        decreaseButton.click();
+
+        expect(numberField.value).to.be.equal('-6');
+      });
+    });
+
+    describe('min and max values are defined', () => {
+      it('should set value to the closest to the max when min is below zero and max is below zero and increaseButton is clicked', () => {
+        numberField.min = -20;
+        numberField.max = -3;
+        numberField.step = 6;
+
+        increaseButton.click();
+
+        expect(numberField.value).to.be.equal('-8');
+
+        // Check with max that can be equally divided
+        numberField.value = '';
+        numberField.min = -24;
+        numberField.step = 6;
+
+        increaseButton.click();
+
+        expect(numberField.value).to.be.equal('-6');
+      });
+
+      it('should set value to 0 when max is zero, min is zero and decreaseButton or increaseButton is clicked', () => {
```

Comment:
~~**idea:** Not sure if it makes anything better, however using `MIN` instead of `min` and `MAX` instead of `max` might facilitate understanding all the circumstances of the test. E.g:~~

~~`should set the value to 0 when MAX = 0, MIN = 0 and a control button is clicked`~~

**Update:** This part is just a copy-paste from the old version of the NumberField.